### ISSUE TITLE
TypeError: Cannot read property 'prev' of null

### ISSUE THREAD

**Issue by dbellavista** [ternjs/tern#891]
Hi!
I'm using tern `0.21.0` on a MacOs system, feeding it with the vim plugin deoplete-ternjs.

I got this error during the autocompletion:

```
/usr/local/lib/node_modules/tern/lib/infer.js:988
        var scopeCopy = new Scope(scope.prev, scope.originNode);
                                       ^

TypeError: Cannot read property 'prev' of null
    at /usr/local/lib/node_modules/tern/lib/infer.js:988:40
    at withDisabledComputing (/usr/local/lib/node_modules/tern/lib/infer.js:359:18)
    at fn.computeRet (/usr/local/lib/node_modules/tern/lib/infer.js:984:14)
    at ctor.addType (/usr/local/lib/node_modules/tern/lib/infer.js:382:18)
    at withWorklist (/usr/local/lib/node_modules/tern/lib/infer.js:908:19)
    at exports.AVal.propagate (/usr/local/lib/node_modules/tern/lib/infer.js:94:25)
    at Modules.<anonymous> (/usr/local/lib/node_modules/tern/plugin/commonjs.js:71:17)
    at Modules.signal (/usr/local/lib/node_modules/tern/lib/signal.js:28:49)
    at exports.Server.<anonymous> (/usr/local/lib/node_modules/tern/plugin/modules.js:338:24)
    at exports.Server.signal (/usr/local/lib/node_modules/tern/lib/signal.js:28:49)
```

**Comment by chemzqm** [comment]
It should be problem of deoplete-ternjs, you can use official vim plugin https://github.com/ternjs/tern_for_vim and bind ominicomplete to deoplete.

**Comment by marijnh** [comment]
I really doubt this has anything to do with deoplete-ternjs. It looks like an issue with the inference engine.

Do you have an example of a file that triggers this error?

**Comment by dbellavista** [comment]
@marijnh I noticed the error only after a while. I'll try to reproduce it but I can't make any promise :)

**Comment by camdenb** [comment]
I'm having this error consistently. Running YouCompleteMe with tern on Vim 8.0.

**Comment by jsinai** [comment]
I am seeing this problem in atom, consistently.

**Comment by lyjstudy** [comment]
FIXED: https://github.com/lyjstudy/tern
rmScope will destroy node.scope, so we cache it

**Comment by marijnh** [comment]
Solved via #905
